universal/include/userver/formats/universal/common_containers.hpp

```diff
@@ -0,0 +1,296 @@
+  using kType = std::unordered_map<std::string, Value>;
+  template <typename MainClass, auto I, typename Value2>
+  inline constexpr kType Read(Value2&& value) const {
+    if(!this->Additional) {
+      throw std::runtime_error("Invalid Flags");
+    }
+    kType response;
+    constexpr auto fields = boost::pfr::names_as_array<MainClass>();
+    for(const auto& [name, value2] : userver::formats::common::Items(std::forward<Value2>(value))) {
+      auto it = std::find(fields.begin(), fields.end(), name);
+      if(it == fields.end()) {
+        response.emplace(name, value2.template As<Value>());
+      }
+    }
+    return response;
+  }
+  template <typename MainClass, auto I, typename Value2>
+  inline constexpr std::optional<kType> TryRead(Value2&& value) const {
+    if(!this->Additional) {
+      throw std::runtime_error("Invalid Flags");
+    }
+    kType response;
+    constexpr auto fields = boost::pfr::names_as_array<MainClass>();
+    constexpr auto name = boost::pfr::get_name<I, MainClass>();
+    for(const auto& [name2, value2] : userver::formats::common::Items(std::forward<Value2>(value))) {
+      if(std::find(fields.begin(), fields.end(), name2) == fields.end()) {
+        auto New = parse::TryParse(value2, parse::To<Value>{});
+        if(!New) {
+          return std::nullopt;
+        };
+        response.emplace(name, *New);
+      }
+    }
+    return response;
+  }
+  inline constexpr std::string_view Check(const kType&) const {
+    return "";
+  }
+  constexpr auto Write(const kType& value, std::string_view, const auto&, auto& builder) const {
+    for(const auto& [name, value2] : value) {
+      builder[name] = value2;
+    };
+  };
+};
+template <typename Element>
+struct FieldConfig<std::vector<Element>> {
+  std::optional<MinElements> MinimalElements;
+  FieldConfig<Element> Items;
+  template <typename MainClass, auto I, typename Value>
+  inline constexpr auto Read(Value&& value) const {
+    constexpr auto name = boost::pfr::get_name<I, MainClass>();
+    std::vector<Element> response;
+    for(const auto& element : value[name]) {
+      auto New = element.template As<Element>();
+      response.push_back(std::move(New));
+    }
+    return response;
+  }
+  template <typename MainClass, auto I, typename Value>
+  inline constexpr std::optional<std::vector<Element>> TryRead(Value&& value) const {
+    std::vector<Element> response;
+    constexpr auto name = boost::pfr::get_name<I, MainClass>();
+    for(const auto& element : value[name]) {
+      auto New = parse::TryParse(element, parse::To<Element>{});
+      if(!New) {
+        return std::nullopt;
+      }
+      response.push_back(std::move(*New));
+    }
+    return response;
+  }
+  inline constexpr std::string Check(const std::vector<Element>& obj) const {
+    std::string error;
+    for(const auto& element : obj) {
+      error += impl::UniversalCheckField(element, this->Items);
+    }
+    return error;
+  }
+
+};
+} // namespace formats::universal
```

Comment:
Let's also support universal serialization for `std::variant`. It should be a class that hides the internal `tuple`, like `SerializationConfig`, with defaulted child `FieldConfig`s by default. The child configs should be able to be overridden by type or by index (for complex cases). Index-based getter should be enough